playground/src/pages/CodingEnvironment.tsx

```diff
@@ -110,18 +115,21 @@ export default function CodingEnvironment(): JSX.Element {
    * It runs the Rust program that is imported above.
    */
   const proveProgram = async () => {
+    toast.loading('Proving ...', {id: 'provingToast'});
```

Comment:
The spinner is not spinning, it seems the proving is running on the rendering thread, so the whole webpage becomes unresponsive. I tested this with the fibonnaci example by changing from `1000` to `100000`.

I think the calls to the WASM code should be moved to a worker instead.

Reply:
Let's have a call about that, I don't understand directly